### PR TITLE
Update logout endpoint to be per project on Masthead

### PR DESCRIPTION
**Pull request by nixocio**
Update logout endpoint to be per project on Masthead

### CODE REVIEW

**Comment by nixocio** [comment]
I talked with @jamestalton and down the road, we may abstract the Masthead component to receive those project specific details as props or another solution, right now adding this one as to allow log out using Hub, and to keep traceability when changing this later on.

File: frontend/common/Masthead.tsx [hunk 1]
```diff
@@ -74,6 +75,10 @@ function isEdaServer(): boolean {
   return process.env.UI_MODE === 'EDA';
 }
 
+function isHubServer(): boolean {
+  return process.env.UI_MODE === 'HUB';
```
Comment (keithjgrant) [inline frontend/common/Masthead.tsx:79]: I'd like to move away from accessing the env variables in the components. I'd suggest passing in the api URLs via a prop. Each app can pass this in where the Masthead is rendered in AwxMain, HubMain, etc

**Review by keithjgrant — CHANGES_REQUESTED** [review]

File: frontend/common/Masthead.tsx (hunk 1 above)
Comment (jamestalton) [reply frontend/common/Masthead.tsx:79]: Im working on abstracting the masthead and making it composable.
That way we can have an AwxMasthead that contains a menu that has a logout that is AWX specific. etc...

File: frontend/hub/api/utils.tsx [hunk 2]
```diff
@@ -8,6 +8,11 @@ import {
   putHubRequest,
 } from './request';
 
+export function getBaseAPIPath() {
```
Comment (jamestalton) [inline frontend/hub/api/utils.tsx:11]: Why do we need getBaseAPIPath when there is hubAPI?

**Review by jamestalton — CHANGES_REQUESTED** [review]

File: frontend/common/Masthead.tsx (hunk 1 above)
Comment (nixocio) [reply frontend/common/Masthead.tsx:85]: @keithjgrant Thanks for your review, as James mentioned there is a long term goal to refactor the masthead to achieve what you described.

File: frontend/hub/api/utils.tsx (hunk 2 above)
Comment (nixocio) [reply frontend/hub/api/utils.tsx]: Updated.

**Review by jamestalton — APPROVED** [review]
lgtm

**Review by keithjgrant — APPROVED** [review]

**Pull request merged**